mifosng-provider/src/main/java/org/mifosplatform/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java

```diff
@@ -1511,7 +1511,7 @@ public LoanTermVariationsData mapRow(final ResultSet rs, @SuppressWarnings("unus
                 .append(" where ((ls.fee_charges_amount <> if(ls.accrual_fee_charges_derived is null,0, ls.accrual_fee_charges_derived))")
                 .append(" or ( ls.penalty_charges_amount <> if(ls.accrual_penalty_charges_derived is null,0,ls.accrual_penalty_charges_derived))")
                 .append(" or ( ls.interest_amount <> if(ls.accrual_interest_derived is null,0,ls.accrual_interest_derived)))")
-                .append("  and loan.loan_status_id=? and mpl.accounting_type=? and loan.is_npa=0 and ls.duedate <= CURDATE() order by loan.id,ls.duedate");
+                .append("  and (loan.loan_status_id=? or loan.expected_maturedon_date >= CURDATE() or loan.expected_maturedon_date > IFNULL(loan.accrued_till, '1900-01-01')) and mpl.accounting_type=? and loan.is_npa=0 and ls.duedate <= CURDATE() order by loan.id,ls.duedate");
```

Comment:
This shouldn't be handled here. Should handle while payments or closure of loan. Once loan is closed there is no point in accruing the income.